src/key.rs

```diff
@@ -30,7 +30,23 @@ use crate::tools::{self, time_elapsed};
 pub(crate) trait DcKey: Serialize + Deserializable + PublicKeyTrait + Clone {
     /// Create a key from some bytes.
     fn from_slice(bytes: &[u8]) -> Result<Self> {
-        Ok(<Self as Deserializable>::from_bytes(Cursor::new(bytes))?)
+        let res = <Self as Deserializable>::from_bytes(Cursor::new(bytes));
+        if let Ok(res) = res {
+            return Ok(res);
+        }
+
+        for garbage_bytes in 3..std::cmp::min(bytes.len(), 10) {
```

Comment:
there should be a brief comment above why this for-loop is done
and also a " info" to the log that this private key has garbage at the end.  
The people verifying the nightly/release can then check if they have this line.

Reply:
oh, so no way to leave a trace?  Can still be merged then but would prefer some user-detectable feedback.

Reply:
Seems that in such cases we can log at least to stderr, why not?